packages/javascript/bh-shared-ui/src/views/Explore/ContextMenu/AssetGroupMenuItemPrivilegeZonesEnabled.tsx

```diff
@@ -16,95 +16,153 @@
+import { Permission, apiClient } from '../../../utils';
+
+const ConfirmNodeChangesDialog: FC<{
+    dialogContent: string;
+    disableAccept: boolean;
+    onAccept: () => void;
+    onCancel: () => void;
+    open: boolean;
+}> = ({ dialogContent, disableAccept, onAccept, onCancel, open }) => {
+    return (
+        <Dialog open={open}>
+            <DialogPortal>
+                <DialogContent>
+                    <DialogTitle>Confirm Selection</DialogTitle>
+                    <DialogDescription>{dialogContent}</DialogDescription>
+                    <DialogActions>
+                        <Button variant='tertiary' onClick={onCancel}>
+                            Cancel
+                        </Button>
+                        <Button variant='primary' onClick={onAccept} disabled={disableAccept}>
+                            Ok
+                        </Button>
+                    </DialogActions>
+                </DialogContent>
+            </DialogPortal>
+        </Dialog>
+    );
+};
+
+export const AssetGroupMenuItem: FC<{
+    assetGroupType: 'tierZero' | 'owned';
     showConfirmationOnAdd?: boolean;
-    confirmationOnAddMessage?: string;
-}> = ({
-    assetGroupId,
-    assetGroupName,
-    disableAddNode,
-    isCurrentMember,
-    removeNodePath,
-    onAddNode = () => {},
-    onShowConfirmation = () => {},
-    onCancelConfirmation = () => {},
-    showConfirmationOnAdd = false,
-    confirmationOnAddMessage = '',
-}) => {
-    const handleAddNode = () => {
-        onAddNode(assetGroupId);
-    };
-
-    const handleOnCancel = () => {
-        onCancelConfirmation();
-    };
-
-    // selected node is not a member of the group
-    if (!isCurrentMember) {
-        if (showConfirmationOnAdd) {
-            return (
-                <>
-                    <MenuItem onClick={onShowConfirmation}>Add to {assetGroupName}</MenuItem>
-                    <ConfirmNodeChangesDialog
-                        onCancel={handleOnCancel}
-                        onAccept={handleAddNode}
-                        dialogContent={confirmationOnAddMessage}
-                        disableAccept={disableAddNode}
-                    />
-                </>
-            );
-        } else {
-            return <MenuItem onClick={handleAddNode}>Add to {assetGroupName}</MenuItem>;
-        }
-    } else {
+}> = ({ assetGroupType, showConfirmationOnAdd = false }) => {
+    const [dialogOpen, setDialogOpen] = useState(false);
+    const { addNotification } = useNotifications();
+    const getAssetGroupTagsQuery = useTagsQuery();
+    const { selectedItemQuery } = useExploreSelectedItem();
+    const { checkPermission } = usePermissions();
+
+    const tierZeroAssetGroup = getAssetGroupTagsQuery.data?.find(
+        (value) => value.position === HighestPrivilegePosition
+    );
+    const ownedAssetGroup = getAssetGroupTagsQuery.data?.find((value) => value.type === AssetGroupTagTypeOwned);
```

Comment:
Unsure if this will remain after my other comment but there are prebuilt hooks that will determine this for you in the useAssetGroupTags hook. While naming things is hard, PZ has tried to differentiate from the old system by using the `Tag` Suffix so in this case something like ownedAssetGroupTag which would help anyone entering this space or searching for AGT related logic